lib/Dialect/FIRRTL/Transforms/AdvancedLayerSink.cpp

```diff
@@ -111,9 +111,18 @@ class EffectInfo {
     });
   }
 
+  /// Record whether the module-like op contains any effectful op.
   void update(FModuleLike moduleOp) {
+    // If the module op has any annotations, then we pretend the module contains
+    // some kind of important effect, so that we cannot sink its instances.
     if (!AnnotationSet(moduleOp).empty())
       return markEffectful(moduleOp);
+
+    for (auto annos : moduleOp.getPortAnnotations()) {
+      if (!AnnotationSet(cast<ArrayAttr>(annos)).empty())
+        return markEffectful(moduleOp);
+    }
```

Comment:
Mega-nit: 
```suggestion
    for (auto annos : moduleOp.getPortAnnotations())
      if (!AnnotationSet(cast<ArrayAttr>(annos)).empty())
        return markEffectful(moduleOp);
```

Suggestion (the `AnnotationSet` is going to be thrown away): 

```suggestion
    for (auto annos : moduleOp.getPortAnnotations()) {
      if (!cast<ArrayAttr>(annos).empty())
        return markEffectful(moduleOp);
    }
```